aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StoragePathGetPresignedUrlOperation.kt

```diff
@@ -48,6 +48,24 @@ internal class AWSS3StoragePathGetPresignedUrlOperation(
                 return@submit
             }
 
+            if (request.validateObjectExistence) {
+                try {
+                    storageService.validateObjectExists(serviceKey)
+                } catch (se: StorageException) {
```

Comment:
I like the way the StoragePath operation is written more than the key one.

Instead of checking for StorageException, as suggested in another comment, lets remove the try/catch on the service call and check for "Not Found" exception here. I think we can also improve the backup exception case which is likely no network. The message can be "Encountered an issue while validating the existence of object" vs "Encountered an issue while generating pre-signed URL"

Reply:
good suggestions, code updated